src/core_plugins/kibana/public/visualize/listing/create_or_delete_button.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+import { CreateButtonLink } from 'ui_framework/components/button/create_button_link';
+import { DeleteButton } from 'ui_framework/components/button/delete_button';
+
+export function CreateOrDeleteButton({ showCreate, doDelete }) {
```

Comment:
Next step would probably be to throw this into ui_framework and pass in createHref and objectType, so other types can re-use it.

I don't love how there isn't symmetry between the create and delete process.  e.g. one is an href one is an action.